homeassistant/components/freebox.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for Freebox devices (Freebox v6 and Freebox mini 4K).
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/freebox/
+"""
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.components.discovery import SERVICE_FREEBOX
+from homeassistant.const import CONF_HOST, CONF_PORT, EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers import config_validation as cv, discovery
+from homeassistant.helpers.discovery import async_load_platform
+
+REQUIREMENTS = ['aiofreepybox==0.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "freebox"
+DATA_FREEBOX = DOMAIN
+
+FREEBOX_CONFIG_FILE = 'freebox.conf'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.port
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the Freebox component."""
+    conf = config.get(DOMAIN)
+
+    async def discovery_dispatch(service, discovery_info):
+        if conf is None:
+            _LOGGER.info("Discovered Freebox server: %s:%s",
+                         discovery_info.get('properties',
```

Comment:
Don't do these look ups twice. Save the values in variables first.

Reply:
I think this is fine as fix